lib/streamlit/elements/heading.py

```diff
@@ -67,15 +71,12 @@ def header(
 
         """
         header_proto = HeadingProto()
-        if anchor is not None:
-            header_proto.anchor = anchor
-        header_proto.body = clean_text(body)
-        header_proto.tag = HEADER_TAG
+        marshall(header_proto, body=body, anchor=anchor, tag=HEADER_TAG)
         return self.dg._enqueue("heading", header_proto)
 
     @gather_metrics("subheader")
     def subheader(
-        self, body: SupportsStr, anchor: Optional[str] = None
+        self, body: SupportsStr, anchor: Union[None, str, Literal[False]] = None
```

Comment:
Since it's a second occurrence of this rather complex type maybe if would be worth creating a type variable?

Reply:
I created a variable.